lib/modules/versioning/pvp/index.ts

```diff
@@ -0,0 +1,328 @@
+  if (versionIntMajor.length === otherIntMajor.length) {
+    return 'eq';
+  }
+  if (versionIntMajor.length > otherIntMajor.length) {
+    return 'gt';
+  }
+  return 'lt';
+}
+
+function isGreaterThan(version: string, other: string): boolean {
+  const versionIntMajor = extractAllComponents(version);
+  const otherIntMajor = extractAllComponents(other);
+  if (versionIntMajor === null || otherIntMajor === null) {
+    return false;
+  }
+  return compareIntArray(versionIntMajor, otherIntMajor) === 'gt';
+}
+
+function getMajor(version: string): number | null {
+  // This basically can't be implemented correctly, since
+  // 1.1 and 1.10 become equal when converted to float.
+  // Consumers should use isSame instead.
+  const components = getComponents(version);
+  if (components === null) {
+    return null;
+  }
+  return Number(components.major.join('.'));
+}
+
+function getMinor(version: string): number | null {
+  const components = getComponents(version);
+  if (components === null || components.minor.length === 0) {
+    return null;
+  }
+  return Number(components.minor.join('.'));
+}
+
+function getPatch(version: string): number | null {
+  const components = getComponents(version);
+  if (components === null || components.patch.length === 0) {
+    return null;
+  }
+  return Number(components.patch[0] + '.' + components.patch.slice(1).join(''));
+}
+
+function matches(version: string, range: string): boolean {
+  const parsed = parseRange(range);
+  if (parsed === null) {
+    return false;
+  }
+  const ver = extractAllComponents(version);
+  const lower = extractAllComponents(parsed.lower);
+  const upper = extractAllComponents(parsed.upper);
+  if (ver === null || lower === null || upper === null) {
+    return false;
+  }
+  return (
+    'gt' === compareIntArray(upper, ver) &&
+    ['eq', 'lt'].includes(compareIntArray(lower, ver))
+  );
+}
+
+function satisfyingVersion(
+  versions: string[],
+  range: string,
+  reverse: boolean,
+): string | null {
+  const copy = versions.slice(0);
+  copy.sort((a, b) => {
+    const multiplier = reverse ? 1 : -1;
+    return sortVersions(a, b) * multiplier;
+  });
+  const result = copy.find((v) => matches(v, range));
+  return result ?? null;
+}
+
+function getSatisfyingVersion(
+  versions: string[],
+  range: string,
+): string | null {
+  return satisfyingVersion(versions, range, false);
+}
+
+function minSatisfyingVersion(
+  versions: string[],
+  range: string,
+): string | null {
+  return satisfyingVersion(versions, range, true);
+}
+
+function isLessThanRange(version: string, range: string): boolean {
+  const parsed = parseRange(range);
+  if (parsed === null) {
+    return false;
+  }
+  const compos = extractAllComponents(version);
+  const lower = extractAllComponents(parsed.lower);
+  if (compos === null || lower === null) {
+    return false;
+  }
+  return 'lt' === compareIntArray(compos, lower);
+}
+
+export function getComponents(splitOne: string): Components | null {
```

Comment:
please move util functions to `./common.ts` or `./util.ts`so they're not part of public versioning API